File: frontend/src/types/account.ts
export interface AccountSummaryData {
  cash: number;
  total_value: number;
  realized_pnl: number | null;
  unrealized_pnl?: number;
  daily_order_count?: number;
  daily_loss_limit_reached?: boolean;
}

export interface Position {
  symbol: string;
  name: string;
  quantity: number;
  average_price: number;
  current_price: number;
  evaluation_pnl?: number;
  return_rate?: number;
}

File: backend/app/schemas/account.py
from decimal import Decimal

from pydantic import BaseModel


class AccountSummary(BaseModel):
    cash: Decimal = Decimal("0")
    total_value: Decimal = Decimal("0")
    realized_pnl: Decimal | None = None
    unrealized_pnl: Decimal = Decimal("0")
    daily_order_count: int = 0
    daily_loss_limit_reached: bool = False


class Position(BaseModel):
    symbol: str
    name: str = ""
    quantity: int
    average_price: Decimal
    current_price: Decimal
    evaluation_pnl: Decimal = Decimal("0")
    return_rate: Decimal = Decimal("0")

